scipy/stats/_censored_data.py

```diff
@@ -0,0 +1,356 @@
+import numpy as np
+
+
+def _validate_1d(a, name):
+    if np.ndim(a) != 1:
+        raise ValueError(f'{name} must be a one-dimensional sequence')
+    if np.isnan(a).any():
+        raise ValueError(f'{name} must not contain nan.')
+    if np.isinf(a).any():
+        raise ValueError(f'{name} must contain only finite values')
+
+
+def _validate_intervals(intervals):
+    intervals = np.asarray(intervals)
+    if intervals.shape == (0,):
+        # The input was a sequence with length 0.
+        intervals = intervals.reshape((0, 2))
```

Comment:
Codecov says these aren't tested.

Reply:
I added a test with empty arrays in https://github.com/scipy/scipy/pull/13699/commits/1d4086ac1c7f5cb8307da5d57da584df7f198a5e